batchflow/models/tf/mobilenet.py

```diff
@@ -219,8 +222,203 @@ def block(cls, inputs, filters, residual=False, strides=1, expansion_factor=6, w
+        config['loss'] = 'ce'
+
+        return config
+
+    def build_config(self, names=None):
+        config = super().build_config(names)
+        config['head']['filters'][-1] = self.num_classes('targets')
+        return config
+
+    @classmethod
+    def body(cls, inputs, name='body', **kwargs):
+        """ Base layers
+
+        Parameters
+        ----------
+
+        inputs : tf.Tensor
+            input tensor
+        name : str
+            scope name
+
+        Returns
+        -------
+        tf.Tensor
+        """
+
+        kwargs = cls.fill_params('body', **kwargs)
+        width_factor, layout = cls.pop(['width_factor', 'layout'], kwargs)
+
+        with tf.variable_scope(name):
+            x = inputs
+            i = 0
+            for block in layout:
+                repeats = block.pop('repeats')
+                block['width_factor'] = width_factor
+                for k in range(repeats):
+                    if k > 0:
+                        block['strides'] = 1
+                    residual = block.pop('residual', False) or k > 0
+                    x = cls.block(x, **block, residual=residual, name='block-%d' % i, **kwargs)
+                    i += 1
+        return x
+
+    @classmethod
+    def block(cls, inputs, filters, residual=False, strides=1, expansion_factor=4, width_factor=1, kernel_size=3,
+              se_block=False, name=None, **kwargs):
+        """ An inverted residual bottleneck block consisting of a separable depthwise convolution and 1x1 pointise
+        convolution and an optional Squeeze-and-Excitation block.
+
+        Parameters
+        ----------
+        inputs : tf.Tensor
+            input tensor
+        filters : int
+            number of output filters
+        residual : bool
+            whether to make a residual connection
+        strides : int
+            stride for 3x3 convolution
+        expansion_factor : int
+            multiplier for the number of filters in internal convolutions
+        width_factor : float
+            multiplier for the number of filters
+        h_swish : bool
+            whether to use hard swish activation. If False tf.nn.relu is used.
+        se_block : bool
+            whether to include squeeze and excitation block
+        name : str
+            scope name
+
+        Returns
+        -------
+        tf.Tensor
+        """
+
+        num_filters = int(cls.num_channels(inputs, kwargs.get('data_format')) * expansion_factor * width_factor)
+        x = conv_block(inputs, 'cna', num_filters, 1, name='%s-exp' % name,
+                       strides=1, **kwargs)
+        x = depthwise_conv(x, kernel_size=kernel_size, strides=strides,
+                           padding='same', data_format=kwargs['data_format'], name='%s-depthwise' % name)
+        x = conv_block(x, 'na', name='%s-na' % name, **kwargs)
+
+        if se_block:
+            x = cls.se_block(x, num_filters // 4, name='%s-se' % name, data_format=kwargs['data_format'],
```

Comment:
Sometimes you use `kwargs['data_format']`, but otherwise `kwargs.get('data_format')`. Be consistent

Reply:
ok